returnn/tf/layers/base.py

```diff
@@ -2434,7 +2434,7 @@ def init(self, output, output_with_activation=None, target=None, layer=None):
             output_with_activation.x, self.output_seq_lens, time_major=output.is_time_major)
         else:
           self.output_flat = flatten_or_merge(output.placeholder, self.output_seq_lens, time_major=output.is_time_major)
-          self.output_flat.set_shape(tf.TensorShape(output.shape))
+          self.output_flat.set_shape(tf.TensorShape((None,) + output.shape[1:]))
```

Comment:
It might be wrong when time dim or batch dim >= 2. But this is anyway not currently supported here...